unittests/FEXLinuxTests/CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+include(ExternalProject)
+ExternalProject_Add(FEXLinuxTests
+  PREFIX FEXLinuxTests
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests"
+  BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/FEXLinuxTests"
+  CMAKE_ARGS
+    "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+    "-DX86_C_COMPILER:STRING=${X86_C_COMPILER}"
```

Comment:
Instead of hardcoding the C/C++ compilers everywhere we cross-compile, how about adding CMake toolchain file to the FEX root directory instead? That allows for easier user-configuration in case they want to use a differently named cross-compiler or in case they want to use different compile flags.

Reply:
> Whatever change we do on this front should be synchronized with how thunks are compiled, as we don't want two different mechanisms for the same thing. I'd make a follow up investigation ticked for that though.

Yeah, agreed this should be done consistently with thunks and is fine for a followup.

> Having said that, I'm far from a CMake expert, and I'm not particularly keen on becoming one. I think we need to switch to a better built system as soon as possible.

This is worth having a deeper discussion at some point, but FTR I have strong reservations against switching away from CMake for FEX, and I don't think doing so would be a net benefit. CMake is not the ideal build system for well-established reasons, but considering some of FEX's strong requirements (using several external dependencies, mixing native builds with cross-compiled projects, ...), I consider it the least terrible option there's currently available.

A cleanup pass over FEX's current CMake files certainly has a lot of potential though.

Reply:
> I consider it the least terrible option there's currently available.

That's how I see it too, right now. We are quite invested in it, swamped by other tasks, and I'm not certain other build systems will fare better :|

> A cleanup pass over FEX's current CMake files certainly has a lot of potential though.
Probably that is a better mid-term goal. Also, reviewing all the helper scripts we use for testing, etc.